extensions/oauth/src/main/java/org/mule/extension/oauth2/internal/OAuthCallbackServersManager.java

```diff
@@ -108,27 +109,47 @@ public boolean isStopped() {
     }
 
     @Override
-    public RequestHandlerManager addRequestHandler(PathAndMethodRequestMatcher pathAndMethodRequestMatcher,
-                                                   RequestHandler requestHandler) {
+    public RequestHandlerManager addRequestHandler(Collection<Method> methods, String path, RequestHandler requestHandler) {
       synchronized (registeredHandlerMatchers) {
-        if (!registeredHandlerMatchers.containsKey(pathAndMethodRequestMatcher)) {
+        if (!registeredHandlerMatchers.containsKey(methods.toString() + "-" + path)) {
```

Comment:
Might be better to have a helper method that calculates the key than concatenating in two places.